### PR TITLE
Add support for Linuxbrew

### LINKED ISSUE #501: Linuxbrew Support
Hi - would a PR be accepted to add Linuxbrew support?

Thanks!

### PR DESCRIPTION
**Pull request by dsully**
First pass at adding Linuxbrew support.

Use `on_macos`/`on_linux` and `on_arm`/`on_intel` as Homebrew stanzas instead of `Hardware::CPU.type`

Update insta snaps.

Fixes #501

### CODE REVIEW

**Review by mistydemeo — COMMENTED** [review]
Thank you very much! I'll do a more thorough review tomorrow, but I really appreciate the contribution.

File: cargo-dist/src/tasks.rs [hunk 1]
```diff
@@ -1332,16 +1332,34 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         let hint = format!("brew install {}", install_target);
         let desc = "Install prebuilt binaries via Homebrew".to_owned();
 
-        // If they have an x64 macos build but not an arm64 one, add a fallback entry
-        // to try to install x64 on arm64 and let rosetta2 deal with it.
+        // If they have an x64 macOS build but not an arm64 one, add a fallback entry
+        // to try to install x64 on arm64 and let Rosetta2 deal with it.
```
Comment (mistydemeo) [inline cargo-dist/src/tasks.rs:1336]: If we're going to change the formatting of these, let's use the official spelling here and elsewhere in the file.
```suggestion
        // to try to install x64 on arm64 and let Rosetta 2 deal with it.
```

File: cargo-dist/templates/installer/homebrew.rb.j2 [hunk 2]
```diff
@@ -5,26 +5,56 @@ class {{ formula_class }} < Formula
...
+  on_macos do
+    {#- If arm64/x86_64 builds are the same, skip the Hardware::CPU.type if statement #}
+    {%- if arm64_macos.id == x86_64_macos.id %}
+    url "{{ inner.base_url }}/{{ arm64_macos.id }}"
+    {%- if arm64_macos_sha256 %}
+    sha256 "{{ arm64_macos_sha256 }}"
+    {%- endif %}
+    {%- else %}
+    {%- if arm64_macos.id %}
+    on_arm
```
Comment (mistydemeo) [inline cargo-dist/templates/installer/homebrew.rb.j2:20]: This needs a `do` as part of the block.

```suggestion
    on_arm do
```

File: cargo-dist/templates/installer/homebrew.rb.j2 [hunk 3]
```diff
@@ -5,26 +5,56 @@ class {{ formula_class }} < Formula
...
+      url "{{ inner.base_url }}/{{ arm64_macos.id }}"
+      {%- if arm64_macos_sha256 %}
+      sha256 "{{ arm64_macos_sha256 }}"
+      {%- endif %}
+    end
+    {%- endif %}
+    {%- if x86_64_macos.id %}
+    on_intel
```
Comment (mistydemeo) [inline cargo-dist/templates/installer/homebrew.rb.j2:28]: Likewise.

```suggestion
    on_intel do
```

**Review by mistydemeo — CHANGES_REQUESTED** [review]
Looks good, thank you! Just a few minor comments.

Again, I really appreciate the contribution.

**Comment by dsully** [comment]
Thanks for catching those missing `do` statements. Updated along with the name fixes.

**Review by mistydemeo — COMMENTED** [review]
Looks good, thank you!

We're cutting a new release soon, so I'll get this merged after that. Do you mind rebasing on `main` and updating the test fixtures with the template change you made in the last commit?

**Comment by dsully** [comment]
Done, just pushed another commit.

**Review by mistydemeo — APPROVED** [review]
Great, thank you! I'll get this merged after 0.5.0 is released.

**Comment by mistydemeo** [comment]
Thanks again! We'll include this in a future release, most likely 0.6.0.

**Pull request merged**